lib/sles4sap_publiccloud.pm

```diff
@@ -667,4 +673,96 @@ sub create_hana_vars_section {
     return (\%hana_vars);
 }
 
+=head2 get_hana_database_info
+
+    Parses "hdbsql" command output.
+    Returns 1 if hana database is online otherwise 0 offline.
+
+=back
+=cut
+
+sub get_hana_database_info {
+    my ($self, $password_db, $instance_id) = @_;
+    my $hana_database_online = '"HDB","HDB-$instance_id","YES","ACTIVE","","","DEFAULT","?"';
+
+    sleep 30;
+    eval {
+        my $output_cmd = $self->run_cmd(cmd => "hdbsql -u SYSTEM -p $password_db -i $instance_id 'SELECT * FROM SYS.M_DATABASES;'", runas => get_required_var("SAP_SIDADM"));
+    };
+    if ($@) {
+        record_info('HANA DB OFFLINE', "Hana database in primary node is offline. Here the error \n$@");
+        return 0;
+    }
+    else {
+        return 1;
+    }
+}
+
+=head2 is_hana_database_offline
+
+    Check if hana DB is offline.
+    Returns if hana database is online with 1 and offline with 0
+
+=over 1
+
+=item B<TIMEOUT> - default 900
+
+=back
+=cut
+
+sub is_hana_database_offline {
+    my ($self, %args) = @_;
+    my $timeout = bmwqemu::scale_timeout($args{timeout} // 900);
+    my $start_time = time;
+    my $db_status = -1;
+    my $consecutive_passes = 0;
+    my $password_db = "Linux1234";
+    my $instance_id = get_required_var('INSTANCE_ID');
+    my $total_consecutive_passes = ($args{total_consecutive_passes} // 5);
+
```

Comment:
You can prepare `hdbsql` command once

```suggestion
my $hdb_cmd = "hdbsql -u SYSTEM -p $password_db -i $instance_id 'SELECT * FROM SYS.M_DATABASES;'";
```

And later use it like
```suggestion
record_info("Hana database after timeout", $self->run_cmd(cmd => $hdb_cmd));
```

Reply:
thx